crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -122,6 +122,19 @@ impl<T: EntityMapper> DynEntityMapper for T {
     }
 }
 
+/// Helper to construct an [`EntityMapper`] from a [`DynEntityMapper`].
+pub struct UnDynEntityMapper<'a>(pub &'a mut dyn DynEntityMapper);
+
+impl<'a> EntityMapper for UnDynEntityMapper<'a> {
+    fn map_entity(&mut self, entity: Entity) -> Entity {
+        self.0.dyn_map_entity(entity)
+    }
+
+    fn mappings(&self) -> impl Iterator<Item = (Entity, Entity)> {
+        self.0.dyn_mappings().into_iter()
+    }
+}
```

Comment:
I don't love this, but I'm not sure there's a better solution short of splitting `EntityMapper` into two traits, which felt more intrusive.

Reply:
Could you implement the trait on the `&dyn DynEntityMapper` directly?

Reply:
Cool, that works and cleans this up nicely! I always forget that `&dyn` types can themselves have traits implemented on them.